### ISSUE TITLE
Allow commenting on maps

### ISSUE THREAD

**Issue by IgorEliezer** [umap-project/umap#768]
I'd like to use uMap as a tool to get general public feedback.

For example, I create a map about city street pavement. People who visit the map could be allowed to leave comments on the map, or directly on the objects, about street conditions/smoothness or to report potholes.

I as map author could create a standard form with a few options for the visitors to use.

**Comment by Binnette** [comment]
Hi @IgorEliezer, by experience I know that people won't use their computer to report pavement issues or write about road condition.... just because when they use their computer, they are not outside walking on the road. I encourage you to think about a 'mobile' solution that people can use on their phone. So a "form" or a "simple map" seems not to be a good idea for me.

I recommend you to give a try to [StreetComplete](https://github.com/westnordost/StreetComplete). This is an Android app, with simply 2 tap on your phone, you can set the pavement of the street you are walking on. This app is a must have, it have many 'quests' to improve OSM data. Please give it a try in real condition (on a road :wink:) and think again about your real application need :wink: 

Have a nice day. I hope that you will enjoy my advises :+1:

And about "Allow commenting on maps", we will need to consider it technically. Note that people can already add note directly on OSM.

**Comment by yohanboniface** [comment]
I think it's more or less #52

**Comment by IgorEliezer** [comment]
> I recommend you to give a try to StreetComplete. This is an Android app, with simply 2 tap on your phone, you can set the pavement of the street you are walking on. This app is a must have, it have many 'quests' to improve OSM data. Please give it a try in real condition (on a road wink) and think again about your real application need

I know the StreetComplete, but the map is not for OSM mappers or for OSM mapping, but for people, say, who only know how to use Facebook, Whatsapp and Google Maps, people who are over 50 year old, have no desire to learn mapping or cartography and have 5th grade. That is, regular citizens with mundane needs.

If I show them a thematic map -- such as, pavement, street lighting, bus route, land use, where the schools or public clinics are located, where the sewage is thrown --, the best approach would be allowing them to comment on the very map they are interacting with. If I ask them to install an app => I lose 50% feedback. If I ask to open another site and have to teach them how to use osm.org => 99% feedback loss. Because regular people can't afford to volunteer to mapping or even have mind to install an app or to do something different than they are used to.

I think open maps could be an aid to get people involved with city and master plan discussions through Internet and social media. People prefer to shout and curse the mayor on Facebook than to use a specialized app. I want to put an interactive map where they are shouting and cursing.

I've posted uMap on Facebook, all people can do is to post "nice", "the map is missing a speed bump somewhere near the street X" and give likes. I'd like to let comment on them not for the sake of openstreetmap but for public interest discussions.

**Comment by VladimirSlavik** [comment]
Aside from the whole RFE for comments, I think one issue could be, for uMap people need some account. The provider list does not have Facebook, so would that even work? I mean, people would have to leave anonymous comments.

Also, OSM notes are an existing place to put comments for "make the map better this way". That's not necessarily a problem as such, just pointing out this exists...

Finally... wow, it's been a long time @IgorEliezer!

**Comment by IgorEliezer** [comment]
Hello @VladimirSlavik :)

> I think one issue could be, for uMap people need some account. (...) I mean, people would have to leave anonymous comments.

Pretty much like what OSM does, I don't think it is necessary to require an account to leave comment/feedback. Actually, it would be more restrict, since the map author can review and remove unwanted feedbacks from their own maps, while OSM depends on the whole community to report them.

**Comment by akelai** [comment]
I agree it would be great to allow comments on the spatial objects (not just the whole map).

For the whole map, or possibly for each spatial object in some way, maybe it could be possible to use [Remark42](https://github.com/umputun/remark42):
> Remark42 is a self-hosted, lightweight, and simple (yet functional) comment engine, which doesn't spy on users. It can be embedded into blogs, articles or any other place where readers add comments.

**Comment by yohanboniface** [comment]
cf #52 too

**Comment by iriman** [comment]
That's a good idea, and I feel the need for that, too. 
For me a form in umap to submit a note on OSM is also appropriate. Which contains a link to the umap map, so the author can track comments related to the map. Something like onosm.org.
Currently on the map I created for wholemeal bread bakeries, I linked to create new note in description of the elements on the map and for items that are not on the map I've referred users to the wiki (!) in a help menu :) (will anyone comment?)
